moment_kinetics/src/coordinates.jl

```diff
@@ -129,40 +129,145 @@ struct coordinate{T <: AbstractVector{mk_float}}
 end
 
 """
+    get_coordinate_input(input_dict, name)
+
+Read the input for coordinate `name` from `input_dict`, setting defaults, etc.
+"""
+function get_coordinate_input(input_dict, name; ignore_MPI=false)
```

Comment:
Why do we have to supply name and input_dict as separate options? Maybe it is obvious. I think name would have previously been inside the grid_input struct, and we would have passed just the inputs for the specific coordinate of interest, not the input_dict for all coordinates (and then slice with name).

Reply:
`set_defaults_and_get_section!()` is called inside `get_coordinate_input()`, and those are the arguments it expects. `set_defaults_and_get_section!()` does actually require `input_dict`, so it can save the name of the section as being 'expected'.

Reply:
So this is just because you wanted to use an existing function? To me it seems that `set_defaults_and_get_section!()` doesn't have exactly the right functionality for the task here (we should be passing in inputs for a single coordinate, we shouldn't need to check that the data exists, we are passing it!).